zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,81 @@
+use bulletproofs::BulletproofGens;
+use curve25519_dalek::scalar::Scalar;
+use hex;
+use merlin::Transcript;
+
+use zkvm::*;
+
+fn issue_contract(
+    qty: u64,
+    flv: Scalar,
+    issuance_pred: &Predicate,
```

Comment:
You are passing borrow of Predicate, but making a clone of it below. It's recommended to do the opposite: _move_ the value in and not clone. Then the user has a choice: if they do not need to reuse the instance, they can move it it without extra cloning.

https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control

<img width="774" alt="image" src="https://user-images.githubusercontent.com/698/53197493-35e1be80-35cf-11e9-87d8-364c0550943e.png">